pylint/lint/pylinter.py

```diff
@@ -693,13 +694,12 @@ def check(self, files_or_modules: Sequence[str]) -> None:
                 fileitems = self._iterate_file_descrs(files_or_modules)
                 data = None
 
-        # The contextmanager also opens all checkers and sets up the PyLinter class
         with augmented_sys_path(extra_packages_paths):
+            # 2) Get the AST for each FileItem
+            ast_per_fileitem = self._get_asts(fileitems, data)
```

Comment:
I wonder if we have unintended effects from not getting these within the context manager..